lib/src/specs/expression.dart

```diff
@@ -189,10 +195,106 @@ abstract class Expression implements Spec {
         'await',
       );
 
+  /// Returns the result of `++this`.
+  Expression operatorUnaryPrefixIncrement() =>
+      BinaryExpression._(_empty, expression, '++', addSpace: false);
+
+  /// Return the result of `this++`.
+  Expression operatorUnaryPostfixIncrement() =>
+      BinaryExpression._(expression, _empty, '++', addSpace: false);
+
+  /// Returns the result of `-this`.
+  Expression operatorUnaryMinus() =>
+      BinaryExpression._(_empty, expression, '-', addSpace: false);
+
+  /// Returns the result of `--this`.
+  Expression operatorUnaryPrefixDecrement() =>
+      BinaryExpression._(_empty, expression, '--', addSpace: false);
+
+  /// Return the result of `this--`.
+  Expression operatorUnaryPostfixDecrement() =>
+      BinaryExpression._(expression, _empty, '--', addSpace: false);
+
+  /// Returns the result of `this` `&` [other].
+  Expression operatorBitwiseAnd(Expression other) =>
+      BinaryExpression._(expression, other, '&');
+
+  /// Returns the result of `this` `|` [other].
+  Expression operatorBitwiseOr(Expression other) =>
+      BinaryExpression._(expression, other, '|');
+
+  /// Returns the result of `this` `^` [other].
+  Expression operatorBitwiseXor(Expression other) =>
+      BinaryExpression._(expression, other, '^');
+
+  /// Returns the result of `~this`.
+  Expression operatorUnaryBitwiseComplement() =>
+      BinaryExpression._(_empty, expression, '~', addSpace: false);
+
+  /// Returns the result of `this` `<<` [other].
+  Expression operatorShiftLeft(Expression other) =>
+      BinaryExpression._(expression, other, '<<');
+
+  /// Returns the result of `this` `>>` [other].
+  Expression operatorShiftRight(Expression other) =>
+      BinaryExpression._(expression, other, '>>');
+
+  /// Returns the result of `this` `>>>` [other].
+  Expression operatorShiftRightUnsigned(Expression other) =>
+      BinaryExpression._(expression, other, '>>>');
+
   /// Return `{this} = {other}`.
   Expression assign(Expression other) =>
       BinaryExpression._(this, other, '=', isConst: isConst);
 
+  /// Return `this` += [other].
+  Expression assignAdd(Expression other) =>
```

Comment:
@srawlins - can you double check that the naming patterns LGTY? Are there any existing (short) names for these compound assignment operators that you think we should switch to?

Reply:
@natebosch @srawlins I wasn't too sure whether to go with
```dart
Expression assignAdd(Expression other);
```
or
```dart
Expression addAssign(Expression other);
```

However looking at https://processing.org/reference/addassign.html I see that they refer to `+=` as `add assign`, so naming the operators from left to right.

I can change the names if desired.

Reply:
Looking at Rust https://doc.rust-lang.org/std/ops/trait.AddAssign.html I probably should rename them
| from | to | 
| --- | --- |
| `Expression.assignAdd` | `Expression.addAssign` |
| `Expression.assignSubtract` | `Expression.subtractAssign` |
| `Expression.assignMultiply` | `Expression.multiplyAssign` |
| `Expression.assignDivide` | `Expression.divideAssign` |
| `Expression.assignIntDivide` | `Expression.intDivideAssign` |
| `Expression.assignEuclideanModulo` | `Expression.euclideanModuloAssign` |
| `Expression.assignShiftLeft` | `Expression.shiftLeftAssign` |
| `Expression.assignShiftRight` | `Expression.shiftRightAssign` |
| `Expression.assignShiftRightUnsigned` | `Expression.shiftRightUnsignedAssign` |
| `Expression.assignBitwiseAnd` | `Expression.bitwiseAndAssign` |
| `Expression.assignBitwiseXor` | `Expression.bitwiseXorAssign` |
| `Expression.assignBitwiseOr` | `Expression.bitwiseOrAssign` |

Reply:
^^ done

Reply:
I don't think analyzer APIs have anything to add here. We call assignments like `+=` a "compound assignment", but don't have a specific term for `+=` vs `-=`. Those symbols are distinguished by the TokenTypes, which also don't really make an attempt to name anything, beyond exactly what the characters are. For example [PLUS_EQ](https://pub.dev/documentation/analyzer/latest/dart_ast_token/TokenType/PLUS_EQ-constant.html), and [TILDE_SLASH_EQ](https://pub.dev/documentation/analyzer/latest/dart_ast_token/TokenType/TILDE_SLASH_EQ-constant.html).